### PR TITLE
Issue40 - Integrate Add Items Drawer with Database Insert

### PR DESCRIPTION
**Pull request by andrewtchan**
### Pull Request Summary

Integrated AddItemsButton component with item drawer and add item modal to save added items to database. Submit button also sends emails. Refactored relevant buttons to MUI button. Note: edit added items in drawer functionality not yet implemented (as noted in comments of AddItemsButton.txt). 

### Pull Request Checklist

- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [x] The developer name is specified
- [x] The summary is completed
- [x] Ask for a review in communication channels

### Reviewer Checklist - PULL REQUEST REVIEWER ONLY

IMPORTANT: The rest of the sections in this checklist should only be filled out by authorized pull request reviewers. If you are the individual contributor, do not fill out the rest of the fields or check the boxes.

## Reviewer: Full Name

NOTE: Pull requests should only be merged when all boxes are checked.

- [ ] Assign yourself as reviewer if not assigned
- [ ] The reviewer name is specified
- [ ] The code is fully reviewed
- [ ] Meaningful feedback is given
- [ ] Let developer know a review has been made

### CODE REVIEW

File: src/components/addItems/AddItemButton.tsx [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
 import "../../styles/AddItemButton.css";
 
-function AddItemButton() {
+function AddItemButton({
+  isDisabled,
+  rows,
+  setRows,
+  setIsDrawerOpen,
+}: {
+  isDisabled: boolean;
```
Comment (jlaksana) [inline src/components/addItems/AddItemButton.tsx:15]: It is cleaner to create a new Typescript type that defines the proptypes and then setting it here. See Table.tsx

File: src/components/addItems/AddItemButton.tsx [hunk 2]
```diff
@@ -1,27 +1,39 @@
...
+  rows: GridRowsProp;
+  setRows: React.Dispatch<React.SetStateAction<GridRowsProp>>;
+  setIsDrawerOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
   const [email, setEmail] = useState("");
 
   // Sample data for testing
```
Comment (jlaksana) [inline src/components/addItems/AddItemButton.tsx:22]: Don't leave any "dead" code!

File: src/components/addItems/AddItemButton.tsx [hunk 3]
```diff
@@ -76,10 +88,10 @@ function AddItemButton() {
   async function handleClick() {
     // Handle a request to add a list of items
```
Comment (jlaksana) [inline src/components/addItems/AddItemButton.tsx:89]: Let'd do some data validation here before submitting. Check that the email value is a valid email. Can look up the common regex for it. Also check it isn't empty. You can add a new error state that becomes true when the email is invalid and then makes the input field red.

**Review by jlaksana — COMMENTED** [review]
Please go through the comments and delete/change the outdated ones. I added the functionality to edit a row.

File: src/components/addItems/AddItemButton.tsx (hunk 3 above)
Comment (andrewtchan) [reply src/components/addItems/AddItemButton.tsx:90]: I gave this a try and did some testing. Now the submit button is disabled until valid items are in the drawer and the email passes the regex. The email input field is red until a valid email is entered. Please let me know if this is what we wanted.

**Review by jlaksana — APPROVED** [review]

**Pull request merged**